opentelemetry-api/src/opentelemetry/propagation/__init__.py

```diff
@@ -0,0 +1,288 @@
+
+    TODO: update docs to reflect split into extractor/injector
+
+    This class provides an interface that enables extracting and injecting
+    span context into headers of HTTP requests. HTTP frameworks and clients
+    can integrate with HTTPTextFormat by providing the object containing the
+    headers, and a getter and setter function for the extraction and
+    injection of values, respectively.
+
+    Example::
+
+        import flask
+        import requests
+        from opentelemetry.context.propagation import Extractor
+
+        PROPAGATOR = HTTPTextFormat()
+
+
+
+        def get_header_from_flask_request(request, key):
+            return request.headers.get_all(key)
+
+        def set_header_into_requests_request(request: requests.Request,
+                                             key: str, value: str):
+            request.headers[key] = value
+
+        def example_route():
+            span_context = PROPAGATOR.extract(
+                get_header_from_flask_request,
+                flask.request
+            )
+            request_to_downstream = requests.Request(
+                "GET", "http://httpbin.org/get"
+            )
+            PROPAGATOR.inject(
+                span_context,
+                set_header_into_requests_request,
+                request_to_downstream
+            )
+            session = requests.Session()
+            session.send(request_to_downstream.prepare())
+
+
+    .. _Propagation API Specification:
+       https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-propagators.md
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def extract(
+        cls,
+        carrier: ContextT,
+        context: typing.Optional[Context] = None,
+        get_from_carrier: typing.Optional[Getter[ContextT]] = None,
+    ) -> Context:
+        """Create a Context from values in the carrier.
+
+        The extract function should retrieve values from the carrier
+        object using get_from_carrier, use values to populate a
+        Context value and return it.
+
+        Args:
+            carrier: and object which contains values that are
+                used to construct a Context. This object
+                must be paired with an appropriate get_from_carrier
+                which understands how to extract a value from it.
+            context: The Context to read values from.
+            get_from_carrier: a function that can retrieve zero
+                or more values from the carrier. In the case that
+                the value does not exist, return an empty list.
+        Returns:
+            A Context with configuration found in the carrier.
+        """
+
+
+class Injector(abc.ABC):
+    @classmethod
+    @abc.abstractmethod
+    def inject(
+        cls,
+        carrier: ContextT,
+        context: typing.Optional[Context] = None,
+        set_in_carrier: typing.Optional[Setter[ContextT]] = None,
+    ) -> None:
```

Comment:
According to the OTEP it should return the carrier. Do you think it should return it? I'm able to understand the motivation of returning them.

Reply:
That's a good question. Looking at the examples, it looks like the [simplified API](https://github.com/open-telemetry/oteps/blob/master/text/0066-separate-context-propagation.md#simplify-the-api-with-automated-context-propagation) injects directly into the carrier. but the previous example shows it setting the request header to the return value.